File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/composite/Leaf.java
package jv.com.tencent.cain.design_pattern.composite;

/**
 * @author cainjiang
 * @date 2020/5/7
 */
public class Leaf extends Component {
    public static final String TAG = "Leaf";

    public Leaf() {
    }

    public Leaf(String name) {
        super(name);
    }

    @Override
    public void add(Component component) {
        System.out.println(TAG + "#" + "add# " + "can't add componet to leaf.");
    }

    @Override
    public void remove(Component component) {
        System.out.println(TAG + "#" + "remove# " + "can't remove componet from leaf.");
    }

    @Override
    public void display(int depth) {
        StringBuilder depthBuilder = new StringBuilder();
        for (int i = 0; i < depth; i++) {
            depthBuilder.append("-");
        }
        depthBuilder.append(name);
        System.out.println(depthBuilder);
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/composite/Leaf.kt
package kt.com.tencent.cain.desigin_pattern.composite

/**
 * @author cainjiang
 * @date 2020/5/7
 */
class Leaf(name:String) : Component(name) {
    companion object {
        const val TAG = "Leaf"
    }

    override fun add(component: Component) {
        println(TAG + "#add# can't add componet to leaf.")
    }

    override fun remove(component: Component) {
        println(TAG + "#add# can't remove componet from leaf.")
    }

    override fun display(depth: Int) {
        val depthBuilder = StringBuilder()
        for (i in 0 until depth) {
            depthBuilder.append("-")
        }
        depthBuilder.append(name)
        println(depthBuilder)
    }
}